Aligns add mood functionality with new project description and converted it to a dialog fragment

mauliksamit opened the pull request:
resolves #176
resolves #177
resolves #179
resolves #180 
resolves #192
resolves #197
resolves #199

mauliksamit commented:
test file is not finalized as @lndslpn appears to be working on it.

ArmaanKatyal commented:
> test file is not finalized as @lndslpn appears to be working on it.

Can you also add the tests for the new stuff you added. afaik she is fixing the old tests not the new additions

mauliksamit commented:
> > test file is not finalized as @lndslpn appears to be working on it.
> 
> Can you also add the tests for the new stuff you added. afaik she is fixing the old tests not the new additions

yup the new stuff was added

ArmaanKatyal commented on code/app/src/main/res/drawable/rounded_background.xml:
```diff
@@ -1,5 +1,5 @@
 <shape xmlns:android="http://schemas.android.com/apk/res/android">
 <corners android:radius="27dp"/> <!-- Adjust radius for desired roundness -->
-<solid android:color="#78766a"/>
+<solid android:color="#5a5a5a"/>
```
why the color was changed?

mauliksamit replied:
It's the colour of the input field boxes. 

The new colour is now consistent with the colour of the input boxes in the filter fragment. It looked a bit off before.

ArmaanKatyal commented on code/app/src/main/res/drawable/spinner_rounded_background_with_arrow.xml:
```diff
@@ -2,8 +2,8 @@
-            <corners android:radius="27dp" /> <!-- Match the radius from rounded_background -->
-            <solid android:color="#242323" /> <!-- Match your background color or use a transparent color -->
+            <corners android:radius="27dp" />
+            <solid android:color="#5a5a5a" />
```
color?

mauliksamit replied:
same as above

ArmaanKatyal commented on code/app/src/main/res/layout/fragment_add_mood_event.xml:
```diff
@@ -1,170 +1,190 @@
-<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:app="http://schemas.android.com/apk/res-auto"
-    xmlns:tools="http://schemas.android.com/tools"
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
why are we still using linearLayout. Can we switch to constraint layout. Thanks!

animeshm19 replied:
+1

animeshm19 commented on code/app/src/main/res/layout/fragment_add_mood_event.xml:
```diff
@@ -1,170 +1,191 @@
+
+    </com.google.android.material.card.MaterialCardView>
+
+</LinearLayout>
```
overall, it looks good to me. just add constraint layout

mauliksamit commented:
Added more tests

Enforced visibility constraints (PUBLIC) for the events that are fetched on the home page

UI tweaks/ Add Mood Event now uses a constraint layout

Added a button to allow the removal of attached images before mood events are saved.

animeshm19 commented:
minor: can you change the textcolor to @color instead of hardcoding hex

mauliksamit commented:
Added visibility material icon on history page.
Slight changes to Add Mood Event test.
Home Page and Map page now only show publicly visible mood events

mauliksamit commented:
resolved merge conflicts

mauliksamit commented:
noice
